testgen/chain.go

```diff
@@ -0,0 +1,306 @@
+package testgen
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Chain is a lightweight blockchain-like store which can read a hivechain
```

Comment:
We could put this into hive, but then we'd actually import from hive here. Not sure this is something we want. We really just need this type as a convenient holder of chain information.

Reply:
We also had a lot more library code before to generate the chain, and this kind of replaces it.

Reply:
I agree, it's more it seems this has use beyond `rpctestgen`. Anyone who wants to build off of `hivechain` in go will need to write something similar. It looks like we are already doing this here and in the devp2p tests for some methods.

Reply:
Are you trying to tell me that something needs to be changed in this PR? I can create the library in hive if you want it that way, just not sure if you're saying I should.

Reply:
I am happy to merge it as-is if you don't want or don't think it should be separate. It just doesn't make sense to me logically to have the package in this repo because rpctestgen is just a consumer of hivechain.

Just let me know if what you prefer!

Reply:
I prefer to just leave it here and merge. It's only 300 LoC and most of it is boring code to load the files. The methods on `Chain` are somewhat specific to RPC tests, we need other functionality in the devp2p tests. We don't even use the same hivechain outputs in devp2p tests. Overall I doubt there is any substantial value here that must be shared with other modules.